src/server/modules/storage/storage.router.ts

```diff
@@ -0,0 +1,36 @@
+import { z } from 'zod'
+import { generateSignedPutUrl } from '~/lib/r2'
+import { env } from '~/server/env'
+import { protectedProcedure, router } from '~/server/trpc'
+
+export const storageRouter = router({
+  presignAvatar: protectedProcedure
+    .input(
+      z.object({
+        fileContentType: z.string(),
+      })
+    )
+    .output(
+      z
+        .object({
+          url: z.string(),
+          key: z.string(),
+        })
+        .or(z.null())
+    )
+    .mutation(async ({ ctx, input: { fileContentType } }) => {
+      if (!env.NEXT_PUBLIC_ENABLE_STORAGE) return null
+      const imageKey = `${ctx.session.user.id}/avatar-${Date.now()}`
```

Comment:
this needed or cloudflare cache will just not update the image until after it expires :(.
drawback is that now all avatars are kept lor, but maybe in the future will have cron job to periodically clear old avatars (should be monotonically increasing)